piv/key_test.go

```diff
@@ -579,3 +580,51 @@ func TestYubiKeyPrivateKeyPINError(t *testing.T) {
 		t.Errorf("expected sign to fail with pin prompt that returned error")
 	}
 }
+
+func TestRetiredKeyManagementSlot(t *testing.T) {
+	type args struct {
+		key uint32
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  Slot
+		want1 bool
+	}{
+		{
+			name:  "Non-existent slot, before range",
+			args:  args{key: 0x0},
+			want:  Slot{},
+			want1: false,
+		},
+		{
+			name:  "First retired slot key",
+			args:  args{key: 0x82},
+			want:  Slot{0x82, 0x5fc10d},
+			want1: true,
+		},
+		{
+			name:  "Last retired slot key",
+			args:  args{key: 0x95},
+			want:  Slot{0x95, 0x5fc120},
+			want1: true,
+		},
+		{
+			name:  "Non-existent slot, after range",
+			args:  args{key: 0x96},
+			want:  Slot{},
+			want1: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := RetiredKeyManagementSlot(tt.args.key)
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
nit: types who's fields are comparable are themselves comparable, you don't need reflection and can just do:

```go
if got != tt.want || gotOK != tt.wantOK {
    t.Errorf(...)
}
```